src/platform/pal_error.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "pal_internal.h"
+#include "atomic_ref_count.h"
+
+namespace xlang
+{
+    namespace impl
+    {
+        struct error_info : xlang_error_info
+        {
+            explicit error_info(xlang_result result) noexcept
+                : m_result(result)
+            {}
+
+            int32_t XLANG_CALL QueryInterface(xlang_guid const& id, void** object) noexcept override
+            {
+                if (id == xlang_unknown_guid)
```

Comment:
The `if (id == xlang_unknown_guid)` and `if (id == xlang_error_info_guid)` branches could be combined. Not a big deal either way though.